tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -850,5 +850,523 @@ int main(int argc, char **argv)
+            }
+
+            /* Chacha20 boosting is disabled when client did not indicate chacha20 preference */
+            {
+                DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+                EXPECT_NOT_NULL(connection);
+
+                connection->security_policy_override = &security_policy;
+                connection->kex_params.server_ecc_evp_params.negotiated_curve = s2n_all_supported_curves_list[0];
+                EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(connection, S2N_TLS12));
+
+                EXPECT_SUCCESS(s2n_connection_set_config(connection, server_config));
+
+                static struct s2n_cipher_suite *test_cipher_suite_list[] = {
+                    /* Negotiated because client did not signal chacha20 boosting and it is present in client wire */
+                    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       
+                    /* Never considered */
+                    &s2n_ecdhe_rsa_with_aes_256_cbc_sha384,         
+                    /* Never considered; if client did signal chacha20 boosting we expect this to be negotiated */
+                    &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256,     
+                };
+
+                cipher_preferences = (struct s2n_cipher_preferences) {
+                    .count = s2n_array_len(test_cipher_suite_list),
+                    .suites = test_cipher_suite_list,
+                    .allow_chacha20_boosting = true,
+                };
+
+                /*
+                 * Client did not signal chacha20 boosting. TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 is not negotiable.
+                 * TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256 > TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256 
+                 */
+                uint8_t aes_128_boosted_wire[] = {
+                    TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+                    TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,
+                    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                };
+                uint8_t count = sizeof(aes_128_boosted_wire) / S2N_TLS_CIPHER_SUITE_LEN;
+
+                /* Verify that client negotiates non-chacha20 ciphersuite when chacha20 boosting is not signalled by client */
+                cipher_preferences.allow_chacha20_boosting = true;
+                EXPECT_SUCCESS(s2n_set_cipher_as_tls_server(connection, aes_128_boosted_wire, count));
+                EXPECT_EQUAL(connection->secure->cipher_suite, &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256);
+            }
+
+            /* Server negotiates its most preferred chacha20 ciphersuite for tls 1.3 */
+            {
+                DEFER_CLEANUP(struct s2n_connection *connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+                EXPECT_NOT_NULL(connection);
+                EXPECT_SUCCESS(s2n_enable_tls13_in_test());
+
```

Comment:
https://github.com/aws/s2n-tls/pull/3543#discussion_r996173747

You should not be enabling/disabling tls13 in every test like this, or at all. These methods shouldn't even affect anything, since you're calling them after creating the config and you're setting your own custom security policy anyway.

TLS1.3 should be enabled/disabled in s2n-tls via security policy. This method is from the initial TLS1.3 development work and is deprecated.